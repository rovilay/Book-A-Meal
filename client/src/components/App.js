import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Header from '../components/common/header';
import TableRow from '../components/common/tablerow';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [
        {
          id: 1,
          name: 'akinola'
        },

        {
          id: 2,
          name: 'ogooluwa'
        },

        {
          id: 3,
          name: 'akinrinade'
        }
      ]
    };
  }
  render() {
    return (
      <div>
        <Header />
        <h1>My React App!</h1>
        <table>
          <tbody>
            {this.state.data.map((person, i) => <TableRow key = {i} data = {person} />)}
          </tbody>
        </table>
      </div>
    );
  }
}

// App.propTypes =  {
//   person: PropTypes.object
// }

export default App;

