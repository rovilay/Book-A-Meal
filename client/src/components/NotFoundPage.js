import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setNav } from '../actions/navLinksActions';

export class NotFoundPage extends Component {
  componentDidMount() {
    this.props.setNav([]);
  }

  render() {
    return (
      <div className="not-found-page-container">
        <div className="not-found-page">
          <h1>404!</h1>
          You seem Lost!
          Click <Link to="/">here</Link> to go home.
        </div>
      </div>
    );
  }
}

NotFoundPage.propTypes = {
  setNav: PropTypes.func.isRequired
};

export default connect('', { setNav })(NotFoundPage);
