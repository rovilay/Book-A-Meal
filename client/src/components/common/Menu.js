import React from 'react';
import PropTypes from 'prop-types';

import MealCard from './MealCard';
import '../../assets/css/mealCard.css';

const Menu = (props) => {
  const { menu } = props;
  return (
<<<<<<< HEAD
    <section className="menu">
      {
        (menu.length > 0)
        &&
        <div className="title" id="title">
          {"Today's Menu"}
        </div>
      }
      {
        (menu.length > 0)
          ?
            <div className="menu-container">
              {
                menu.map(meal => (
                  <MealCard
                    key={meal.id}
                    mealData={meal}
                    {...props}
                  />
                ))
              }
            </div>
          :
            <div className="empty">
                Sorry No Meals Today!
            </div>
=======
    <div className="menu">
      {
        (menu.length > 0)
        &&
        <h1 className="menu-title" id="menu-title">
          Menu
        </h1>
      }
      {
        (menu.length > 0)
        ?
          <div className="menu-container" id="">
            {
              menu.map(meal => (
                <MealCard
                  key={meal.id}
                  mealData={meal}
                  {...props}
                />
              ))
            }
          </div>
        :
          <div className="empty">
              Sorry No Meals Today!
          </div>
>>>>>>> d49f2b57ebed161ee85a43f5828ca57edba505e1
      }
    </section>
  );
};

Menu.propTypes = {
  menu: PropTypes.array.isRequired,
};

export default Menu;
