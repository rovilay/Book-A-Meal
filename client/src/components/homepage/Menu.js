import React, { Component } from 'react';
import PropTypes from 'prop-types';

import MealCard from '../common/MealCard';

class Menu extends Component {
  render() {
    const { menu, onAddMealToCart } = this.props;
    return (
      <div className="menu-container">
        <div className="menu-title" id="menu-title">
          Todays Menus
        </div>
        <div className="boxes" id="menu-display">
          {menu.map((meal, i) => <MealCard key={i} mealData={meal} onSubmit={onAddMealToCart} />)}
        </div>
      </div>
    );
  }
}

Menu.propTypes = {
  menu: PropTypes.array.isRequired,
  onAddMealToCart: PropTypes.func.isRequired
};

export default Menu;

