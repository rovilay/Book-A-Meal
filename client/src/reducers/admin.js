import filterify from '../helpers/filter';

const setDefaultAdminState = {
  meals: [],
  filteredMeals: [],
  setMenuMeals: [],
  editMenuMeals: [],
  menus: [],
  orders: {
    grandTotalPrice: 0,
    history: []
  },
  serverRes: {
    success: '',
    message: ''
  },
  mealOnEdit: {},
  modal: {
    isOpen: false,
    isEdit: false,
    isInfo: false,
    isOrderInfo: false,
    isSetMenu: false,
    close: true,
    content: {},
    contentLable: ''
  }
};

const adminReducer = (state = setDefaultAdminState, action) => {
  switch (action.type) {
    case 'SET_MEALS':
      return {
        ...state,
        meals: [...action.meals]
      };
    case 'SET_ADMIN_MODAL':
      return {
        ...state,
        modal: {
          ...action.modal
        }
      };
    case 'ADD_MEAL_NEW_MENU':
      return {
        ...state,
        setMenuMeals: [...new Set([
          ...state.setMenuMeals,
          action.mealId
        ])]
      };
    case 'REMOVE_MEAL_NEW_MENU':
      return {
        ...state,
        setMenuMeals: [
          ...state.setMenuMeals.filter(id => id !== action.mealId)
        ]
      };
    case 'EMPTY_NEW_MENU':
      return {
        ...state,
        setMenuMeals: []
      };
    case 'SET_MENUS':
      return {
        ...state,
        menus: [
          ...action.menus,
        ]
      };
    case 'SET_DEFAULT':
      return {
        ...setDefaultAdminState
      };
    case 'SERVER_RES':
      return {
        ...state,
        serverRes: { ...action.response }
      };
    case 'RESET_SERVER_RES':
      return {
        ...state,
        serverRes: {
          success: '',
          message: ''
        }
      };
    case 'ADD_MEAL_EDIT_MENU':
      return {
        ...state,
        editMenuMeals: [
          ...new Set([
            ...state.editMenuMeals,
            action.mealId
          ])
        ]
      };
    case 'DELETE_MEAL_EDIT_MENU':
      return {
        ...state,
        editMenuMeals: [
          ...state.editMenuMeals.filter(id => id !== action.mealId)
        ]
      };
    case 'EMPTY_EDIT_MENU':
      return {
        ...state,
        editMenuMeals: []
      };
    case 'DELETE_MEAL_EDIT_MODAL':
      return {
        ...state,
        modal: {
          ...state.modal,
          content: {
            ...state.modal.content,
            meals: [
              ...state.modal.content.meals.filter(meal => meal.id !== action.mealId)
            ]
          }
        }
      };
    case 'SET_MEAL_FOR_EDIT':
      return {
        ...state,
        mealOnEdit: state.meals.filter(meal => meal.id === action.mealId)[0]
      };
    case 'REMOVE_MEAL_FROM_EDIT':
      return {
        ...state,
        mealOnEdit: {}
      };
    case 'SET_ORDERS':
      return {
        ...state,
        orders: {
          ...action.orders
        }
      };
    case 'FILTER_CATERER_MEALS':
      return {
        ...state,
        filteredMeals: filterify(action.filter, state.meals)
      };
    default:
      return state;
  }
};

export default adminReducer;

