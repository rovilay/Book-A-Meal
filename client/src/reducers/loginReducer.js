import { SET_USER_DATA, LOG_OUT_USER } from '../actions/actiontypes';

const userDataDefaultState = {
  user: {
    id: '',
    loginMessage: '',
    isLogin: false,
    firstName: '',
    lastName: '',
    admin: '',
    expire: ''
  }
};

const loginReducer = (state = userDataDefaultState, action) => {
  switch (action.type) {
    case SET_USER_DATA:
      return {
        ...state,
        user: {
          ...action.userData
        }
      };
    case LOG_OUT_USER:
      return { ...userDataDefaultState };
    default:
      return state;
  }
};

export default loginReducer;
