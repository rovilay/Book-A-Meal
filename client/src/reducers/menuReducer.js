import {
  SET_TODAY_MENU,
  SET_ALL_MENUS,
  SET_MENU_MEALS,
  ADD_MEAL_TO_NEW_MENU,
  REMOVE_MEAL_FROM_NEW_MENU,
  DELETE_MEAL_IN_EDIT_MENU,
  DELETE_MEAL_IN_MENU,
  EMPTY_NEW_MENU,
  ADD_MEAL_IN_EDIT_MENU,
  EMPTY_EDIT_MENU,
  SET_MENU_FOR_EDIT
} from '../actions/actiontypes';

export const menuDefaultState = {
  todayMenu: [],
  newMenu: [],
  allMenus: [],
  menuMeals: {
    meals: [],
    pagination: {
      limit: 5,
      offset: 0,
      count: 0,
      numOfPages: 1
    }
  },
  editMenu: [],
  pagination: {
    limit: 10,
    offset: 0,
    count: 0,
    numOfPages: 1
  }
};

export const menuReducer = (state = menuDefaultState, action) => {
  switch (action.type) {
    case SET_TODAY_MENU:
      return {
        ...state,
        todayMenu: action.meals,
        pagination: action.pagination

      };
    case ADD_MEAL_TO_NEW_MENU:
      return {
        ...state,
        newMenu: [...new Set([...state.newMenu, action.mealId])]
      };
    case REMOVE_MEAL_FROM_NEW_MENU:
      return {
        ...state,
        newMenu: state.newMenu.filter(mealId => mealId !== action.mealId)
      };
    case EMPTY_NEW_MENU:
      return {
        ...state,
        newMenu: []
      };
    case SET_ALL_MENUS:
      return {
        ...state,
        allMenus: action.menus,
        pagination: action.pagination
      };
    case SET_MENU_MEALS:
      return {
        ...state,
        menuMeals: action.menuMeals,
        menuMealsPagination: action.pagination
      };
    case DELETE_MEAL_IN_MENU:
      return {
        ...state,
        menuMeals: {
          meals: [...state.menuMeals.meals].filter(
            meal => meal.id !== action.mealId
          ),
          pagination: {
            ...state.menuMeals.pagination,
            count: state.menuMeals.pagination.count - 1
          }
        }
      };
    case SET_MENU_FOR_EDIT:
      return {
        ...state,
        editMenu: action.editMenu
      };
    case ADD_MEAL_IN_EDIT_MENU:
      return {
        ...state,
        editMenu: [...new Set([...state.editMenu, action.mealId])]
      };
    case DELETE_MEAL_IN_EDIT_MENU:
      return {
        ...state,
        editMenu: [
          ...state.editMenu.filter(mealId => mealId !== action.mealId)
        ]
      };
    case EMPTY_EDIT_MENU:
      return {
        ...state,
        editMenu: []
      };
    default:
      return state;
  }
};
