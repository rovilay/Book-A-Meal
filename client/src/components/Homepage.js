import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router';

import { setDefaultNav } from '../actions/navLinksAction';
import getTodayMenu from '../actions/menuAction';
import Menu from './common/Menu';

class HomePage extends Component {
  componentDidMount() {
    this.props.setDefaultNav();
    this.props.getTodayMenu();

    this.addMealToCart = this.addMealToCart.bind(this);
  }

  addMealToCart() {
    this.props.history.push('/login');
  }

  render() {
    const { menu } = this.props;
    return (
      <main>
        <section className="first-section">
          <div className="showcase">
            <p className="merienda"> Meals that perfectly <br />fits your lifestyle</p>
            <a
              role="button"
              href="#menu"
              className="view-menu-btn"
            >
              View menu
            </a>
          </div>
        </section>
        <section className="intro">
          <div className="intro-container">
            <h1 className="color-1">Are you busy and hungry?</h1>
            <p>
              Would you want to eat your favourite food from your favourite resturant?<br />
              Use this app to keep track of our menu and place orders at your convinience.
            </p>
            <p className="buttons">
              <button
                className="btn-1"
                onClick={() => {
                  this.props.history.push('/login');
                }}
              >
                Login
              </button>
              <button
                className="btn-1"
                onClick={() => {
                  this.props.history.push('/signup');
                }}
              >
                Sign up
              </button>
            </p>
          </div>
        </section>
        {
        (menu.length > 0)
        &&
        <Menu
          menu={menu}
          addMealToCart={this.addMealToCart}
          notify={this.notify}
          {...this.props}
        />
      }
      </main>
    );
  }
}

HomePage.propTypes = {
  menu: PropTypes.array.isRequired,
  getTodayMenu: PropTypes.func.isRequired,
  setDefaultNav: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  user: state.login.user,
  menu: state.todayMenu.Meals
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    setDefaultNav,
    getTodayMenu,
  },
  dispatch
);

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HomePage));
