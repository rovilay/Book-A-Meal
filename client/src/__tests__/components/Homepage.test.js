import React from 'react';
import { shallow } from 'enzyme';
import mockStore from '../__mockData__/mockStore';
import { HomePage, mapStateToProps } from '../../components/Homepage';
import { userDataDefaultState } from '../../reducers/loginReducer';
import { menuDefaultState } from '../../reducers/menuReducer';
import { menuMeals } from '../__mockData__/menuMock';
import Menu from '../../components/common/Menu';

describe('Homepage component test', () => {
  const setup = () => {
    const props = {
      pagination: {
        limit: 12,
        count: 13,
        numOfPages: 2
      },
      todayMenu: menuMeals,
      getTodayMenu: jest.fn(),
      history: [],
      setDefaultNav: jest.fn()
    };

    return shallow(<HomePage {...props} />)
  };


  it('should render Homepage correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });


  it('should render connected components', (done) => {
    const props = {
      menu: menuMeals
    };

    const store = mockStore({
      menus: {
        todayMenu: menuMeals
      }
    })

    const wrapper = shallow(<Menu {...props} store={store} />);
    expect(wrapper.length).toBe(1);

    done()
  });


  it('should push to history if either login or sign up button is clicked',
    (done) => {
      const wrapper = setup();

      wrapper.find('button').first().simulate('click');
      wrapper.find('button').last().simulate('click');
      expect(wrapper.instance().props.history.length).toEqual(2);
      expect(wrapper.instance().props.history[0]).toEqual('/login');
      expect(wrapper.instance().props.history[1]).toEqual('/signup');

      done();
  });

  it('should map state to props', () => {
    const initialState = {
      login: {
        ...userDataDefaultState
      },
      menu: {
        ...menuDefaultState
      }
    };

    const ownProps = {
      user: 'ogooluwa',
      todayMenu: menuMeals,
      pagination: {
        limit: 12,
        offset: 0,
        count: 13,
        nextOffset: 12
      }
    };

    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  })

  it('should call `handlePaginationClick` if pagination button is clicked', (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePaginationClickSpy= jest.spyOn(
      wrapper.instance(), 'handlePaginationClick'
    );
    wrapper.instance().handlePaginationClick(data)
    expect(handlePaginationClickSpy).toHaveBeenCalled();

    done();
  });
});
