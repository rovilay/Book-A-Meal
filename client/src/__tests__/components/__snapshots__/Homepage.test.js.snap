// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component test should map state to props 1`] = `
Object {
  "pagination": Object {
    "count": 0,
    "limit": 10,
    "numOfPages": 1,
    "offset": 0,
  },
  "todayMenu": Array [],
  "user": Object {
    "admin": "",
    "expire": "",
    "firstName": "",
    "id": "",
    "isLogin": false,
    "lastName": "",
    "loginMessage": "",
  },
}
`;

exports[`Homepage component test should render Homepage correctly 1`] = `
<main
  className="homepage"
>
  <section
    className="first-section"
  >
    <div
      className="showcase"
    >
      <p
        className="merienda"
      >
        Meals that perfectly fits your lifestyle
      </p>
      <a
        className="view-menu-btn btn-2"
        href="#menu"
        role="button"
      >
        View menu
      </a>
    </div>
  </section>
  <section
    className="intro"
  >
    <div
      className="intro-container"
    >
      <h1
        className="color-1"
      >
        Are you busy and hungry?
      </h1>
      <p>
        Would you want to eat your favourite food from your favourite resturant?
        <br />
        Use this app to keep track of our menu and place orders at your convinience.
      </p>
      <p
        className="buttons"
      >
        <button
          className="btn-2"
          onClick={[Function]}
        >
          Log in
        </button>
        <button
          className="btn-2"
          onClick={[Function]}
        >
          Sign up
        </button>
      </p>
    </div>
  </section>
  <Menu
    getTodayMenu={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    history={Array []}
    menu={
      Array [
        Object {
          "id": "1",
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "2",
          "title": "SALAD",
        },
        Object {
          "id": "3",
          "title": "SPAGHETTI",
        },
        Object {
          "id": "4",
          "title": "YAM AND EGG",
        },
      ]
    }
    pagination={
      Object {
        "count": 13,
        "limit": 12,
        "numOfPages": 2,
      }
    }
    setDefaultNav={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    todayMenu={
      Array [
        Object {
          "id": "1",
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "2",
          "title": "SALAD",
        },
        Object {
          "id": "3",
          "title": "SPAGHETTI",
        },
        Object {
          "id": "4",
          "title": "YAM AND EGG",
        },
      ]
    }
  />
  <div
    className="pagination-container"
  >
    <PaginationBoxView
      activeClassName="active"
      breakClassName="break-me"
      breakLabel={
        <a
          href=""
        >
          ...
        </a>
      }
      containerClassName="pagination"
      disableInitialCallback={false}
      disabledClassName="disabled"
      marginPagesDisplayed={2}
      nextClassName="next"
      nextLabel=">>"
      onPageChange={[Function]}
      pageCount={2}
      pageRangeDisplayed={5}
      previousClassName="previous"
      previousLabel="<<"
      subContainerClassName="pages pagination"
    />
  </div>
</main>
`;
